### PR TITLE
Add support for original year tags

### PR DESCRIPTION
**Pull request by eduardosm**
A song may have a tag with the year of the release of the disc it was ripped from and a tag containing the year of the first release of the disc. 
See "date" and "originaldate": https://picard.musicbrainz.org/docs/tags/

This PR adds support for showing the original year on the playlist and adds "Original year" and "Original year - Album" to the library grouping options.

So far, there is only support for this tag on ID3v2 and Ogg files.

### CODE REVIEW

File: ext/libclementine-tagreader/tagreader.cpp [hunk 1]
```diff
@@ -190,6 +190,13 @@ void TagReader::ReadFile(const QString& filename,
         compilation =
             TStringToQString(map["TCMP"].front()->toString()).trimmed();
 
+      if (!map["TDOR"].isEmpty())
```
Comment (hatstand) [inline ext/libclementine-tagreader/tagreader.cpp]: Use `{}` to make this clearer

File: ext/libclementine-tagreader/tagreader.cpp [hunk 2]
```diff
@@ -313,6 +320,12 @@ void TagReader::ReadFile(const QString& filename,
         Decode(items["\251grp"].toStringList().toString(" "), nullptr,
                song->mutable_grouping());
       }
+
+      if (items.contains("----:com.apple.iTunes:ORIGINAL YEAR"))
```
Comment (hatstand) [inline ext/libclementine-tagreader/tagreader.cpp]: Add a constant for this string with a comment explaining it

File: ext/libclementine-tagreader/tagreader.cpp [hunk 3]
```diff
@@ -353,6 +366,20 @@ void TagReader::ReadFile(const QString& filename,
         }
       }
     }
+
+    if (attributes_map.contains("WM/OriginalReleaseTime")) {
```
Comment (hatstand) [inline ext/libclementine-tagreader/tagreader.cpp]: Add constants and comments for these strings too

**Comment by eduardosm** [comment]
Shall I add declare the constants in tagreader.h and define them in tagreader.c (like kMP4_FMPS_*) or just define them in tagreader.c?

**Comment by hatstand** [comment]
Just define them in `tagreader.cpp` inside an anonymous namespace

**Comment by eduardosm** [comment]
I've just commited the changes. Is it ok?

**Pull request merged**